### PR TITLE
SP-6610 Catch ngModel being null or undefined

### PR DESCRIPTION
**Pull request by dejakob**
http://team.showpad.com:8008/browse/SP-6611
http://master.showpad.dev:8080/job/Colorpicker-SP-6610-colorpicker-broken/

### CODE REVIEW

File: js/src/bootstrap-colorpicker.directive.js [hunk 1]
```diff
@@ -56,7 +56,7 @@
             function transformChangedColor (event)
             {
                 const newColorValue = event.color.toHex().toLowerCase();
-                const oldColorValue = $scope.ngModel.toLowerCase();
+                const oldColorValue = ($scope.ngModel || '').toLowerCase();
```
Comment (klaascuvelier) [inline js/src/bootstrap-colorpicker.directive.js]: I'd prefer this, just for the sake of purity:
````
let oldColorValue = $scope.ngModel;

if (typeof oldColorValue === 'string') {
    oldColorValue = oldColorValue.toLowerCase();
}
````
Or using a ternary expression

File: js/bootstrap-colorpicker-module.js [hunk 2]
```diff
@@ -61,7 +61,13 @@
              */
             function transformChangedColor(event) {
                 var newColorValue = event.color.toHex().toLowerCase();
-                var oldColorValue = $scope.ngModel.toLowerCase();
+                var oldColorValue = $scope.ngModel;
+
+                if (typeof oldColorValue === 'string') {
+                    oldColorValue = oldColorValue.toLowerCase();
+                } else {
+                    oldColorValue = '';
```
Comment (klaascuvelier) [inline js/bootstrap-colorpicker-module.js]: but there is not reason to set this to `''`, you can leave it at `null` or whatever.

File: js/src/bootstrap-colorpicker.directive.js [hunk 3]
```diff
@@ -39,12 +41,22 @@
             $element.colorpicker({ format: transformationSettings.name });
             $element.colorpicker().on('changeColor', transformChangedColor);
 
+            if ($scope.shouldUpdateOnKeyUp) {
```
Comment (klaascuvelier) [inline js/src/bootstrap-colorpicker.directive.js]: Where are you removing this listener?

File: lib/bootstrap-colorpicker.js [hunk 4]
```diff
@@ -189,13 +191,24 @@
...
         color: this.color
       });
+
+      /**
+       * Mousedown event on the document
+       * @param eventData
+       */
+      function onMouseDown (eventData)
+      {
+          if (eventData.target !== self.element[0]) {
```
Comment (klaascuvelier) [inline lib/bootstrap-colorpicker.js]: Sure this completely works as expected? Clicking inside on different elements and so?
Remember the "smarter" way of doing this -> an invisible overlay over the page, just underneath the color picker

File: js/src/bootstrap-colorpicker.directive.js (hunk 3 above)
Comment (dejakob) [inline js/src/bootstrap-colorpicker.directive.js]: In the commit I added now :)

File: lib/bootstrap-colorpicker.js (hunk 4 above)
Comment (dejakob) [inline lib/bootstrap-colorpicker.js]: I wouldn't do that.
As of all logic and design belongs to the bootstrap colorpicker lib, I would rather keep changes as small as possible.
To answer your question: yes, I checked that everything works as expected, only the input box itself is an exception to the rule.

Comment (klaascuvelier) [inline lib/bootstrap-colorpicker.js]: If it works fine, it;s okay for me :)

**Comment by klaascuvelier** [comment]
You forgot your `build` in the last commit

**Comment by dejakob** [comment]
I did not... the bootstrap lib doesn't need a build it is included in the vendors of centralstation...

**Comment by klaascuvelier** [comment]
Okay, in that case, add more comments to that file on top (saying what changes we did)
Because now for me it is not clear this is a copied file with our changes in.

**Pull request merged**